### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by rbrito** [Pat-Carter/stream2chromecast#74]
Hi, @Pat-Carter.

I think that this project is super nice, but it seems that the code could, in general, be improved considerably in some cases.

There are already some pull requests and I have ideas for improving the code even more. If you don't mind, I would like to send you some pull requests.

My question is: Do you want some pull requests to help with the project?

Cheers,

Rogério Brito.

**Comment by Pat-Carter** [comment]
Thanks for taking the time to ask.

To be honest I don't really enjoy maintaining other people's code. I have to do that during the day.

There are always improvements to be made but I have learned over the years that code style improvements for their own sake are never a good idea.

This script was intended to send media to my Chromecast. It currently does that. I'm happy to share it here and, when time allows, to help people out if they have issues.

I have a python 3 version, but I prefer python 2 and will switch only when necessary. I enjoy python 2. Python 3 less so.

You're obviously welcome to fork it if you wish. I'll link to your fork if you like.

**Comment by rbrito** [comment]
Hi, Pat.

On Feb 09 2018, Pat Carter wrote:
> Thanks for taking the time to ask.

Thanks for taking the time to answer. I am sorry that I couldn't reply
earlier, but we had a national holiday and I didn't have proper
connectivity.

> To be honest I don't really enjoy maintaining other people's code. I have
> to do that during the day.

The idea here would be to make it easy to introduce one feature or another
while making the code more Pythonic, nothing too intrusive, I would hope.

> There are always improvements to be made but I have learned over the years
> that code style improvements for their own sake are never a good idea.

No, the idea that I had was not to make style improvements for their own
sake. Some of the changes are correctness changes, like not using mutable
types as default arguments and so on.

> This script was intended to send media to my Chromecast. It currently does
> that. I'm happy to share it here and, when time allows, to help people out
> if they have issues.

That's great and that's useful to other people. That would be awesome.

> I have a python 3 version, but I prefer python 2 and will switch only when
> necessary. I enjoy python 2. Python 3 less so.

The problem is that inclusion of Python 2-only programs is problematic in
Linux distributions, since Python 2 is "near" its death. Python 3, on the
other hand, has its life guaranteed.

That being said, as the author of a very widely used Python program (see
https://github.com/coursera-dl/coursera-dl), I don't feel that much
difference between Python 2 and Python 3 and the code runs equally well on
both versions (and I even use Pypy as an alternative interpreter, just for
fun).

> You're obviously welcome to fork it if you wish. I'll link to your fork if you like. 

I don't really like having forks if I can integrate my changes upstream,
making one version better for everybody.

Furthermore, I can see with your code that you have vast knowledge and
experience with the inner workings of the chromecast protocol, something
that I will only get after interacting with you.

In other words, together we are better than separately.


Hoping you reconsider the option to integrate a few pull requests for
improvements,


Rogério.

-- 
Rogério Brito : rbrito@{ime.usp.br,gmail.com} : GPG key 4096R/BCFCAAAA
http://cynic.cc/blog/ : github.com/rbrito : profiles.google.com/rbrito
DebianQA: http://qa.debian.org/developer.php?login=rbrito%40ime.usp.br

**Comment by Pat-Carter** [comment]
Hi Rogério,

I understand your points and I do sympathise with some of them. Perhaps, I can give my point of view as well, though.

As background, I've been coding since the 1970s, and professionally for more than 30 years. 

I've worked with quite a lot of different languages over that time. I currently like Python as a useful scripting tool, but I don't consider it a religion or an identity. I really don't care if something is Pythonic or not. I'm just not a purist.

I was disappointed by the approach to the changes Python3 and I don't particularly want to use it over Python2. There was no need for it. We already had unicode strings and I see no particular reason why print should be a function.

This was a script I put together to achieve a specific aim for myself and I decided to share it here. I wasn't trying to impress anyone with my skills. I'm happy if people find it useful but I don't consider it to be a responsibility.

I don't get paid to maintain this script and there have been people on here who seem to feel entitled to demand features from me or to have their changes incorporated into my project. I have been a little disappointed by peoples' attitudes on github.

However, I do appreciate that you took the time to ask first - thanks for that.

Regarding Python in Linux - I'd be very surprised if the python command points to anything other Python2 for a very long time. Most of the servers I work on still have only Python2 installed.

When Python2 gets EOLed in a couple of years time, perhaps we should fork it for the sake of humanity?  :-)

Anyway, to conclude - this script was meant to be a one-off, itch-scratching hack - but it got a little out of control.  

PyChromecast is a much more professional project that covers the same sort of area. Also, I understand that the latest version of VLC now streams directly to Chromecasts too (though I haven't tried it yet).

I'm currently in the process of moving house, but I will take another look at those changes soon. However, my rule is, If the change doesn't fix anything and I find it boring, I won't be doing it (unless someone wants to pay me). 

Life is too short to spend working on a script that already works. I have other things that I like to work on in my spare time.

cheers,

Pat
